### PR TITLE
Groupfix

### PR DESCRIPTION
**Pull request by pradpnayak**
https://civicrm.stackexchange.com/questions/46021/civicrm-membership-sync-module-for-backdrop-null-array/46023?noredirect=1#comment54427_46023

### CODE REVIEW

**Comment by argiepiano** [comment]
The fix submitted here is not ideal because it uses the null coalescing operator, which is not available in PHP 5.6, which is still supported by Backdrop.

**Comment by herbdool** [comment]
I'm merging. CiviCRM has a minimum requirement of PHP 7.4 so this PR is fine.

File: modules/civicrm_group_roles/civicrm_group_roles.module [hunk 1]
```diff
@@ -583,7 +583,11 @@ function civicrm_group_roles_add_rule_form_submit($form, &$form_state) {
   // If edit_flag is not set then get next highest number.
   $config = config('civicrm_group_roles.settings');
   if (empty($form_state['values']['edit_flag'])) {
-    $id = max(array_keys($config->get('rules'))) + 1;
+    $id = 1;
+    $rules = ($config->get('rules') ?? []);
+    if (!empty(rules)) {
```
Comment (commonpike) [inline modules/civicrm_group_roles/civicrm_group_roles.module:588]: Is this a typo ? `if (!empty($rules)) {`


also, doesnt this already solve it ?
```
$id = max(array_keys($config->get('rules')??[''])) + 1;
```

Comment (commonpike) [reply modules/civicrm_group_roles/civicrm_group_roles.module:588]: Fixed here

- https://github.com/civicrm/civicrm-backdrop/commit/6d448de05546aa268015960aaed80b641f83b6a2

**Pull request merged**